### DISCUSSION TITLE
Ignoring a click event on a specific element within the reference element

### DISCUSSION THREAD

**Discussion by denchen** [floating-ui/floating-ui#2330] [Q&A] [answered]
I'm trying to implement a multi-select dropdown. The reference element will basically contain all the selected items as a series of chips / tags. Each chip/tag has a small "X" where you can remove just that selected item. Of course, when I click on the item, I don't want the open state to change (ie, if the dropdown is already open, it should remain open, but if it's closed, it should remain closed). I tried to add `e.stopPropagation()`, but that didn't seem to work.

I then noticed that `useFloating()`'s `onOpenChange()` was called before the Chip's `onClick()` handler was called. My expectation would be of the reverse.

Here's the relevant code (removed a bunch of stuff that I don't think matters)

```tsx
const [isOpen, setIsOpen] = useState(false);

const { context } = useFloating({
  onOpenChange: (isOpen) => {
    console.log('onOpenChange');
    setIsOpen(isOpen)
  }
});

const { getFloatingProps, getItemProps, getReferenceProps } =
  useInteractions([
  useClick(context),
  useDismiss(context)
});
      
return <div
  ref={refs.setReference}
  {...getReferenceProps()}
>
  {selectedItems.map(item =>
    <Chip
      item={item}
      onClick={e => {
        console.log('Heard onClick()');
        e.stopPropagation();
        // Logic to remove item here
      })
    />
  }
</div>
```

So with the above, when I click on the Chip, I see "onOpenChange" in the logs before "Heard onClick()"

So the question is, how do I prevent the open/close state change when I click on the chip, yet still allow open/close if I click anywhere else in the reference element?

**Comment by atomiks** [comment]
`onClickCapture` on the `Chip`? A CodeSandbox would help to debug a bit.

Code works fine here: https://codesandbox.io/s/stoic-yonath-jl4rnm?file=/src/App.tsx

**Reply by denchen** [reply]
Thanks. Yours does work, but I don't quite know why mine doesn't. The original code sandbox was taken from one of the official examples and modified

https://codesandbox.io/s/xenodochial-curran-2h14dx?file=/src/App.tsx

**Reply by atomiks** [reply, marked as answer]
Oh that's because it uses `onMouseDown` not `onClick` in this case. So stopping propagation in that event instead works

**Reply by denchen** [reply]
Thanks! I added the `stopPropagation()` to `onMouseDown()`, and I get the behavior I expect.
